File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/BotAddedEvent.java
package work.socialhub.kslack.entity.event;

import com.github.seratch.jslack.api.model.BotIcons;

/**
 * The bot_added event is sent to all connections for a workspace when an integration "bot" is added.
 * Clients can use this to update their local list of bots.
 * <p>
 * If the bot belongs to a Slack app, the event will also include an app_id pointing to its parent app.
 * <p>
 * https://api.slack.com/events/bot_added
 */
public class BotAddedEvent implements Event {

    public static final String TYPE_NAME = "bot_added";

    private final String type = TYPE_NAME;
    private Bot bot;

    public String getType() {
        return this.type;
    }

    public Bot getBot() {
        return this.bot;
    }

    public void setBot(Bot bot) {
        this.bot = bot;
    }

    public static class Bot {
        private String id;
        private String appId;
        private String name;
        private BotIcons icons;

        public String getId() {
            return this.id;
        }

        public String getAppId() {
            return this.appId;
        }

        public String getName() {
            return this.name;
        }

        public BotIcons getIcons() {
            return this.icons;
        }

        public void setId(String id) {
            this.id = id;
        }

        public void setAppId(String appId) {
            this.appId = appId;
        }

        public void setName(String name) {
            this.name = name;
        }

        public void setIcons(BotIcons icons) {
            this.icons = icons;
        }
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/BotAddedEvent.kt
package work.socialhub.kslack.entity.event

import com.github.seratch.jslack.api.model.BotIcons

/**
 * The bot_added event is sent to all connections for a workspace when an integration "bot" is added.
 * Clients can use this to update their local list of bots.
 *
 *
 * If the bot belongs to a Slack app, the event will also include an app_id pointing to its parent app.
 *
 *
 * https://api.slack.com/events/bot_added
 */
class BotAddedEvent : Event {
    override val type: String = TYPE_NAME
    var bot: Bot? = null

    class Bot {
        var id: String? = null
        var appId: String? = null
        var name: String? = null
        var icons: BotIcons? = null
    }

    companion object {
        const val TYPE_NAME: String = "bot_added"
    }
}